ds3-cli-certification/src/test/java/com/spectralogic/ds3cli/certification/Certification_Test_8.java

```diff
@@ -0,0 +1,403 @@
+        boolean success = false;
+
+        OUT.startNewTest(testDescription);
+        try {
+            OUT.insertLog("Set data policy to use versioning");
+            final String enableDataPolicyVersioningCmd = "--http -c modify_data_policy --modify-params versioning:KEEP_LATEST -i " + envDataPolicyId;
+            final CommandResponse modifyDataPolicyResponse = Util.command(client, enableDataPolicyVersioningCmd);
+            OUT.insertCommand(enableDataPolicyVersioningCmd, modifyDataPolicyResponse.getMessage());
+            assertThat(modifyDataPolicyResponse.getReturnCode(), is(0));
+
+            // create and store one file
+            Path bulkPutLocalTempDir = CertificationUtil.createTempFiles(bucketName, numFiles, fileSize);
+            final String putInitialCmd = "--http -c put_bulk -b " + bucketName + " -d " + bulkPutLocalTempDir.toString();
+            final CommandResponse putInitialResponse = Util.command(client, putInitialCmd);
+            OUT.insertCommand(putInitialCmd, putInitialResponse.getMessage());
+            assertThat(putInitialResponse.getReturnCode(), is(0));
+
+            OUT.insertLog("List bucket contents");
+            final String getBucketCmd = "--http -c get_bucket -b " + bucketName;
+            final CommandResponse listInitialContentsResponse = Util.command(client, getBucketCmd);
+            OUT.insertCommand(getBucketCmd, listInitialContentsResponse.getMessage());
+            assertThat(listInitialContentsResponse.getReturnCode(), is(0));
+
+            // make new object half size, clobber previous
+            OUT.insertLog("Put new object of same name but half size");
+            bulkPutLocalTempDir = CertificationUtil.createTempFiles(bucketName, numFiles, fileSize / 2);
+            final String putNewVersionCmd = "--http --force -c put_bulk -b " + bucketName + " -d " + bulkPutLocalTempDir.toString();
+            final CommandResponse putNewVersionResponse = Util.command(client, putNewVersionCmd);
+            OUT.insertCommand(putNewVersionCmd, putNewVersionResponse.getMessage());
+            assertThat(putNewVersionResponse.getReturnCode(), is(0));
+
+            OUT.insertLog("List bucket contents (half size)");
+            final CommandResponse listNewContentsResponse = Util.command(client, getBucketCmd);
+            OUT.insertCommand(getBucketCmd, listNewContentsResponse.getMessage());
+            assertThat(listNewContentsResponse.getReturnCode(), is(0));
+
+            success = true;
+        } catch (final Exception e) {
+            LOG.info("Exception: {}", e.getMessage(), e);
+        } finally {
+            Util.deleteBucket(client, bucketName);
+
+            // undo versioning
+            Util.command(client, "--http -c modify_data_policy --modify-params versioning:NONE -i " + envDataPolicyId);
+
+            OUT.finishTest(testDescription, success);
+            assertTrue(success);
```

Comment:
Consider using this form of the assertTrue statement http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertTrue(java.lang.String, boolean)

In the String portion include some text about what didn't happen, or what went wrong.